src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -698,16 +698,15 @@ def _set_route_table(ncf, resource_group_name, route_table, subnet):
         subnet.route_table = None
 
 def create_subnet(resource_group_name, virtual_network_name, subnet_name,
-                  address_prefix='10.0.0.0/24', network_security_group=None,
+                  address_prefix, network_security_group=None,
                   route_table=None):
-    '''Create a virtual network (VNet) subnet
+    '''Create a virtual network (VNet) subnet.
     :param str address_prefix: address prefix in CIDR format.
     :param str network_security_group: Name or ID of network security
         group to associate with the subnet.
     '''
     ncf = _network_client_factory()
     subnet = Subnet(name=subnet_name, address_prefix=address_prefix)
-    subnet.address_prefix = address_prefix
```

Comment:
For `create subnet` the address prefix is now required, so the user has to specify it. No more default. Line 710 was removed because it was redundant. 

The logic to set the default subnet applies only to `vnet create` and it is in the `_validators.py` file.